api/v1alpha1/etcdcluster_types.go

```diff
@@ -50,10 +52,58 @@ type ProviderConfig struct {
 	CertManagerCfg *ProviderCertManagerConfig `json:"certManagerCfg,omitempty"`
 }
 
+type AltNames struct {
+	// DNSNames is the expected array of DNS subject alternative names.
+	// if empty defaults to $(POD_NAME).$(ETCD_CLUSTER_NAME).$(POD_NAMESPACE).svc.cluster.local
+	// +optional
+	DNSNames []string `json:"dnsNames,omitempty"`
+
+	// IPs is the expected array of IP address subject alternative names.
+	// +optional
+	IPs []net.IP `json:"ipAddresses,omitempty"`
+}
+
+type CommonConfig struct {
+	// CommonName is the expected common name X509 certificate subject attribute.
+	// Should have a length of 64 characters or fewer to avoid generating invalid CSRs.
+	// +optional
+	CommonName string `json:"commonName,omitempty"`
+
+	// Organization is the expected array of Organization names to be used on the Certificate.
+	// +optional
+	Organization []string `json:"organizations,omitempty"`
+
+	// AltNames contains the domain names and IP addresses that will be added
+	// to the x509 certificate SubAltNames fields. The values will be passed
+	// directly to the x509.Certificate object.
+	AltNames AltNames `json:"altNames,omitempty"`
+
+	// ValidityDuration is the expected duration until which the certificate will be valid,
+	// expects in human-readable duration: 100d12h, if empty defaults to 90d
+	// +optional
+	ValidityDuration string `json:"validityDuration,omitempty"`
+
+	// CABundleSecret is the expected secret name with CABundle present
+	// and validate if the Issuer provided has the same secretName
+	// if empty, directly use IssuerKind, IssuerName for certificate creation
```

Comment:
As we discussed, the `CABundleSecret` has nothing to do with the users' certificate management system (i.e. cert-manager), so it has nothing to do with IssuerKind and IssuerName here. It's used by etcd instance to verify the TLS communication in case a CA isn't present in the secret created by cert-manager.

Reply:
Yes, updated the description.